sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -606,6 +607,28 @@ public static <T> TypedRead<T> read(SerializableFunction<SchemaAndRecord, T> par
         .build();
   }
 
+  /**
+   * Reads from a BigQuery table or query and returns a {@link PCollection} with one element per
+   * each row of the table or query result, where the custom {@link org.apache.avro.io.DatumReader}
+   * implementation is used to parse from the BigQuery AVRO format.
+   *
+   * <p> This API allows direct deserialization of AVRO data to the target class.
+   */
+  public static <T> TypedRead<T> readWithDatumReader(
```

Comment:
rather than a new method, can we just add `withDatumReaderFactory` on TypedRead?

Reply:
how would we get the TypedRead in that case? were you thinking something like:
```
 BigQueryIO.readTableRows()
            .withDatumReaderFactory((AvroSource.DatumReaderFactory<User>) (writer, reader) -> new SpecificDatumReader<>(reader))
            .withReaderSchema(User.getAvroSchema())
``` 
? (this doesn't work though)

Reply:
oh hmm good call, maybe we can overload `read` then instead?